components/TextFields/src/Private/MDCTextInputUnderlineView.m

```diff
@@ -0,0 +1,183 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTextInputUnderlineView.h"
+
+#import "MaterialPalettes.h"
+
+NSString *const MDCTextInputUnderlineColorKey = @"MDCTextInputUnderlineColorKey";
+NSString *const MDCTextInputUnderlineEnabledKey = @"MDCTextInputUnderlineEnabledKey";
+NSString *const MDCTextInputUnderlineLineHeightKey = @"MDCTextInputUnderlineLineHeightKey";
+
+static const CGFloat MDCTextInputUnderlineDefaultHeight = 1.f;
+
+// TODO: (larche): Make disabled color parameterized?
+static inline UIColor *MDCTextInputUnderlineColor() {
+  return [UIColor lightGrayColor];
+}
+
+@implementation MDCTextInputUnderlineView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    _color = MDCTextInputUnderlineColor();
+    _enabled = YES;
+    _lineHeight = MDCTextInputUnderlineDefaultHeight;
+
+    [self commonMDCUnderlineViewInitialization];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  self = [super initWithCoder:coder];
+  if (self) {
+    _color = [coder decodeObjectForKey:MDCTextInputUnderlineColorKey];
+    _enabled = [coder decodeBoolForKey:MDCTextInputUnderlineEnabledKey];
+    _lineHeight = (CGFloat)[coder decodeFloatForKey:MDCTextInputUnderlineLineHeightKey];
+
+    [self commonMDCUnderlineViewInitialization];
+  }
+  return self;
+}
+
+- (void)commonMDCUnderlineViewInitialization {
+  [self setContentHuggingPriority:UILayoutPriorityDefaultHigh
+                          forAxis:UILayoutConstraintAxisVertical];
+  [self setClipsToBounds:NO];
+  [self updateUnderline];
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+  [super encodeWithCoder:coder];
+  [coder encodeObject:self.color forKey:MDCTextInputUnderlineColorKey];
+  [coder encodeBool:self.enabled forKey:MDCTextInputUnderlineEnabledKey];
+  [coder encodeFloat:(float)self.lineHeight forKey:MDCTextInputUnderlineLineHeightKey];
+}
+
+- (instancetype)copyWithZone:(NSZone *)zone {
+  MDCTextInputUnderlineView *copy = [[[self class] alloc] initWithFrame:self.frame];
+
+  copy.color = self.color.copy;
+  copy.enabled = self.enabled;
+  copy.lineHeight = self.lineHeight;
+
+  return copy;
+}
+
+- (void)layoutSubviews {
```

Comment:
Do we need to set translatesAutoresizingMaskIntoConstraints = NO on initFromCoder since we have a custom implementation?